TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementCheckers.cs

```diff
@@ -22,4 +23,20 @@ public static class SavepointStatementChecker
             }
         }
     }
+
+    public static class ExecuteImmediateStatementChecker
+    {
+        public static void OnCodeElement(ExecuteImmediateStatement executeImmediateStatement, CodeElementsParser.ExecuteImmediateStatementContext context)
+        {
+            var test = executeImmediateStatement.Expression is HostVariable;
+            if (!test) return;
+            var hostVariable = (HostVariable) executeImmediateStatement.Expression;
+            if (hostVariable.IndicatorReference != null)
+            {
+                DiagnosticUtils.AddError(executeImmediateStatement,
+                    "The host variable mustn't have any indicator variable",
+                    context);
+            }
```

Comment:
No
- you have to check the type of the expression, use ExpressionType property it's faster
- only 2 types are allowed: variable or constant, create a diagnostic if anything else is found
- for variable
  - must be an HostVariable (use VariableType), without indicator
- for constant
  - must be Alphanumeric

Reply:
Sorry about that misleading comment. The variable don't have to be a host variable but when it is it has to have no idictaor.